sqlglot/parser.py

```diff
@@ -1039,6 +1048,11 @@ def _parse_limit(self):
             return None
         return self.expression(exp.Limit, this=self._parse_number())
 
+    def _parse_top(self):
```

Comment:
let's just delete this and add an arg to _parse_limit which is 

def _parse_limit(self, top=False):
self._match(TokenType.TOP if top else TokenType.LIMIT)